pyqtorch/noise.py

```diff
@@ -0,0 +1,83 @@
+from __future__ import annotations
+
+from typing import Tuple
+
+import torch
+
+from pyqtorch.matrices import DEFAULT_MATRIX_DTYPE, IMAT, XMAT, _dagger
+from pyqtorch.utils import Operator, State
+
+
+class Noise(torch.nn.Module):
```

Comment:
can we make this class inherit from `Primitive`?

Reply:
Yes, good idea. I will modify this.

Reply:
Finally, I don't know because the inputs of the class are `List[Tensor]`. 
In `Primitive`, the `pauli` variable  is a `Tensor`, so everything has to be redefined in `Noise` .
And I don’t know what to write for `pauli` in `super.__init()__`

Reply:
Is this even going to be differentiable?
If not, I'd say turn off the autodiff features and save a bunch of memory


Also, further to Dominiks comment.
I think (as a followup perhaps) it could be worthwhile to refactor so that we have a class that combines some of the common setup between the primitives class and this one

Reply:
Thank you for all your insightful comments @awennersteen.

Even though the noise gates do not have parameters to optimize directly, they need to be differentiable to allow gradient propagation throughout the circuit. This ensures that the parameters of other parametric gates, like the `RX` gate can be properly optimized. I need to ensure that all operations in `Noise` class are compatible with automatic differentiation to enable consistent and efficient optimization of the entire circuit.


Also, It would indeed be beneficial to create a parent class for both the `Primitive` class and `Noise` class to consolidate common setups. I will consider implementing this in a future PR. Thank you for the idea.